pkg-manager/plugin-commands-installation/src/createJsrParam.ts

```diff
@@ -0,0 +1,28 @@
+import { createJsrPref, parseJsrSpec } from '@pnpm/jsr-specs'
+import { type Dependencies } from '@pnpm/types'
+
+/**
+ * Currently, `pnpm update` relies on recreating CLI params to pass to `installDeps`.
+ *
+ * For `pnpm update` to correctly modify the spec of `jsr:` dependencies, the CLI params must
+ * be bare `jsr:` specification that specifies only scope and name (no version, range, nor tag).
+ *
+ * This function checks if the {@link alias} is a `jsr:` dependency and returns a bare `jsr:` pref
+ * if it is. Otherwise, it returns {@link alias} as-is.
+ */
+export function createJsrParamWithoutSpec (dependencies: Dependencies, alias: string): string {
+  const pref: string | undefined = dependencies[alias]
+  if (pref == null) return alias
+
+  const spec = parseJsrSpec(pref)
```

Comment:
This naming is confusing. The function says that a spec is parsed while the result is also a spec. Meanwhile the input is a pref, not a spec.